static/panes/device-view.ts

```diff
@@ -0,0 +1,378 @@
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import {FontScale} from '../widgets/fontscale';
+import * as monaco from 'monaco-editor';
+import _ from 'underscore';
+import $ from 'jquery';
+import * as colour from '../colour';
+import {ga} from '../analytics';
+import * as monacoConfig from '../monaco-config';
+import {PaneRenaming} from '../widgets/pane-renaming';
+import TomSelect from 'tom-select';
+import GoldenLayout from 'golden-layout';
+import {Hub} from '../hub';
+import {MonacoPane} from './pane';
+import {DeviceAsmCode, DeviceAsmState} from './device-view.interfaces';
+import {MonacoPaneState} from './pane.interfaces';
+
+type DecorationEntry = {
+    linkedCode: any[];
+};
+
+type DeviceType = {
+    languageId: string;
+    asm: DeviceAsmCode[];
+};
+
+export class DeviceAsm extends MonacoPane<monaco.editor.IStandaloneCodeEditor, DeviceAsmState> {
+    private decorations: DecorationEntry;
+    private prevDecorations: string[];
+    private awaitingInitialResults: boolean;
+    private selectedDevice: string;
+    private devices: Record<string, DeviceType> | null;
+    private colours: string[];
+    private deviceCode: DeviceAsmCode[];
+    private lastColours: Record<number, number>;
+    private lastColourScheme: string;
+    private readonly selectize: TomSelect;
+    private linkedFadeTimeoutId: NodeJS.Timeout | null;
+
+    public constructor(hub: Hub, container: GoldenLayout.Container, state: DeviceAsmState & MonacoPaneState) {
+        super(hub, container, state);
+
+        this.prevDecorations = [];
+
+        this.awaitingInitialResults = false;
+        this.selection = state.selection;
+        this.selectedDevice = state.device || '';
+        this.devices = null;
+
+        this.colours = [];
+        this.deviceCode = [];
+        this.lastColours = [];
+        this.lastColourScheme = '';
+
+        const changeDeviceEl = this.domRoot[0].querySelector('.change-device') as HTMLInputElement;
+        this.selectize = new TomSelect(changeDeviceEl, {
+            sortField: 'name',
+            valueField: 'name',
+            labelField: 'name',
+            searchField: ['name'],
+            options: [],
+            items: [],
+            dropdownParent: 'body',
+            plugins: ['input_autogrow'],
+        });
+
+        this.paneRenaming = new PaneRenaming(this, state);
+
+        this.initButtons(state);
```

Comment:
This is called `registerButtons` in the base class and already called in its constructor